pkg/gen/python-templates/CustomResource.py

```diff
@@ -15,37 +19,69 @@ class CustomResource(pulumi.CustomResource):
     `ServiceMonitor` resource definition as an argument.
     """
 
-    def __init__(self, __name__, __opts__=None, api_version=None, kind=None, metadata=None, spec=None):
-        if not __name__:
+    def __init__(
+            self,
+            resource_name: str,
+            api_version: str,
+            kind: str,
+            spec: pulumi.Input[Any] = None,
+            metadata: Optional[pulumi.Input[Any]] = None,
+            opts: Optional[pulumi.ResourceOptions] = None,
+            __name__: str = None,
+            __opts__: pulumi.ResourceOptions = None):
+        if __name__ is not None:
+            warnings.warn("explicit use of __name__ is deprecated", DeprecationWarning)
+            resource_name = __name__
+        if __opts__ is not None:
+            warnings.warn("explicit use of __opts__ is deprecated, use 'opts' instead", DeprecationWarning)
+            opts = __opts__
+        if not resource_name:
             raise TypeError('Missing resource name argument (for URN creation)')
-        if not isinstance(__name__, str):
+        if not isinstance(resource_name, str):
             raise TypeError('Expected resource name to be a string')
-        if __opts__ and not isinstance(__opts__, pulumi.ResourceOptions):
+        if opts and not isinstance(opts, pulumi.ResourceOptions):
             raise TypeError('Expected resource options to be a ResourceOptions instance')
 
         __props__ = dict()
 
-        if not api_version:
-            raise TypeError('Missing required property apiVersion')
         __props__['apiVersion'] = api_version
-        if not kind:
-            raise TypeError('Missing required property kind')
         __props__['kind'] = kind
-        if spec is None:
-            raise TypeError('Missing required property spec')
         __props__['spec'] = spec
         __props__['metadata'] = metadata
 
-        if __opts__ is None:
-            __opts__ = pulumi.ResourceOptions()
-        if __opts__.version is None:
-            __opts__.version = version.get_version()
+        if opts is None:
+            opts = pulumi.ResourceOptions()
+        if opts.version is None:
+            opts.version = version.get_version()
```

Comment:
can you change all four lines above to this:  `opts = pulumi.ResourceOptions.merge(opts, pulumi.ResourceOptions(version=version.get_version()))`?